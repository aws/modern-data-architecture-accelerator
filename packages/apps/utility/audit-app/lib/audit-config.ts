/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParser, CaefAppConfigParserProps, CaefBaseConfigContents } from '@aws-caef/app';
import { BucketInventoryProps } from '@aws-caef/audit-l3-construct';
import { CaefRoleRef } from '@aws-caef/iam-role-helper';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';

import * as configSchema from './config-schema.json';

export interface AuditConfigContents extends CaefBaseConfigContents {
    /**
     * List of refs to roles which will be provided read access to the audit logs
     */
    readRoles?: CaefRoleRef[]
    /**
     * List of source accounts from which audit logs will be accepted
     */
    sourceAccounts?: string[]
    /**
     * List of source regions from which audit logs will be accepted
     */
    sourceRegions?: string[]
    /**
     * Specifies the S3 prefix where inventory data will be accepted on the audit bucket
     */
    inventoryPrefix?: string
    /**
     * The list of expected inventories
     */
    inventories?: BucketInventoryProps[]
}

export class AuditConfigParser extends CaefAppConfigParser<AuditConfigContents> {

    public readonly readRoleRefs: CaefRoleRef[]
    public readonly sourceAccounts: string[]
    public readonly sourceRegions: string[]
    public readonly inventoryPrefix: string
    public readonly inventories?: BucketInventoryProps[]

    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )

        this.readRoleRefs = this.configContents.readRoles ? this.configContents.readRoles : []
        this.sourceAccounts = this.configContents.sourceAccounts ? this.configContents.sourceAccounts : []
        this.sourceRegions = this.configContents.sourceRegions ? this.configContents.sourceRegions : []
        this.inventoryPrefix = this.configContents.inventoryPrefix ? this.configContents.inventoryPrefix : "inventory/"
        this.inventories = this.configContents.inventories
    }

}

