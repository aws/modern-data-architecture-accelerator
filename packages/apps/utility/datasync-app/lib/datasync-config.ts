/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParser, CaefAppConfigParserProps, CaefBaseConfigContents } from '@aws-caef/app';
import { AgentProps, AgentWithNameProps, LocationNfsProps, LocationObjectStorageProps, LocationS3Props, LocationsByTypeWithNameProps, LocationSmbProps, TaskProps, TaskWithNameProps, VpcProps } from '@aws-caef/datasync-l3-construct';

import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';


import * as configSchema from './config-schema.json';

export interface LocationsByTypeConfig {
    readonly s3?: { [ name: string ]: LocationS3Props }
    readonly smb?: { [ name: string ]: LocationSmbProps }
    readonly nfs?: { [ name: string ]: LocationNfsProps }
    readonly objectStorage?: { [ name: string ]: LocationObjectStorageProps }
}

export interface DataSyncConfigContents extends CaefBaseConfigContents {
    readonly vpc?: VpcProps
    readonly agents?: { [ name: string ]: AgentProps }
    readonly locations?: LocationsByTypeConfig
    readonly tasks?: { [ name: string ]: TaskProps }
}

export class DataSyncConfigParser extends CaefAppConfigParser<DataSyncConfigContents> {
    public readonly vpc?: VpcProps
    public readonly agents?: AgentWithNameProps[]
    public readonly locations?: LocationsByTypeWithNameProps
    public readonly tasks?: TaskWithNameProps[]

    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.vpc = this.configContents.vpc
        this.agents = Object.entries( this.configContents.agents || {} ).map( x => {
            return {
                agentName: x[ 0 ],
                ...x[ 1 ]
            }
        } )

        this.locations = {
            s3: Object.entries( this.configContents.locations?.s3 || {} ).map( x => { return { locationName: x[ 0 ], ...x[ 1 ] } } ),
            smb: Object.entries( this.configContents.locations?.smb || {} ).map( x => { return { locationName: x[ 0 ], ...x[ 1 ] } } ),
            nfs: Object.entries( this.configContents.locations?.nfs || {} ).map( x => { return { locationName: x[ 0 ], ...x[ 1 ] } } ),
            objectStorage: Object.entries( this.configContents.locations?.objectStorage || {} ).map( x => { return { locationName: x[ 0 ], ...x[ 1 ] } } )
        }

        this.tasks = Object.entries( this.configContents.tasks || {} ).map( x => {
            return {
                name: x[ 0 ],
                ...x[ 1 ]
            }
        } )
    }
}

