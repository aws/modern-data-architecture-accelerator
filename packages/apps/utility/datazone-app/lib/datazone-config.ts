/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParser, CaefAppConfigParserProps, CaefBaseConfigContents } from '@aws-caef/app';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';


import * as configSchema from './config-schema.json';
import { NamedDomainsProps } from '@aws-caef/datazone-l3-construct';

export interface DataZoneConfigContents extends CaefBaseConfigContents {
    readonly domains: NamedDomainsProps
}

export class DataZoneConfigParser extends CaefAppConfigParser<DataZoneConfigContents> {
    public domains: NamedDomainsProps
    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.domains = this.configContents.domains
    }

}

