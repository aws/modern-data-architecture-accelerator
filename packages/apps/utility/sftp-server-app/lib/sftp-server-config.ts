/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParser, MdaaAppConfigParserProps, MdaaBaseConfigContents } from '@aws-mdaa/app'
import { ServerProps } from '@aws-mdaa/sftp-server-l3-construct';
import { Schema } from 'ajv';

import { Stack } from 'aws-cdk-lib';
import * as configSchema from './config-schema.json';


export interface SftpServerConfigContents extends MdaaBaseConfigContents {
    server: ServerProps
}

export class SftpServerConfigParser extends MdaaAppConfigParser<SftpServerConfigContents> {

    public readonly server: ServerProps

    constructor( stack: Stack, props: MdaaAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.server = this.configContents.server
    }
}

