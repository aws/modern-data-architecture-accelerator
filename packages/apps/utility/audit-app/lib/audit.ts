/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { AuditL3Construct, AuditL3ConstructProps } from '@aws-caef/audit-l3-construct';
import { CaefAppConfigParserProps, CaefCdkApp } from '@aws-caef/app';
import { CaefL3ConstructProps } from '@aws-caef/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { AuditConfigParser } from './audit-config';


export class AuditCDKApp extends CaefCdkApp {
    constructor( props: AppProps = {} ) {
        super( "audit", props )
    }
    protected subGenerateResources ( stack: Stack, l3ConstructProps: CaefL3ConstructProps, parserProps: CaefAppConfigParserProps ) {

        const appConfig = new AuditConfigParser( stack, parserProps )
        const constructProps: AuditL3ConstructProps = {
            ...{
                sourceAccounts: appConfig.sourceAccounts,
                sourceRegions: appConfig.sourceRegions,
                readRoleRefs: appConfig.readRoleRefs,
                bucketInventories: appConfig.inventories,
                inventoryPrefix: appConfig.inventoryPrefix
            }, ...l3ConstructProps
        }

        new AuditL3Construct( stack, "audit", constructProps );
        return [ stack ]

    }
}

