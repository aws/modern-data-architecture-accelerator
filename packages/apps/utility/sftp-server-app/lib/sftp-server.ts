/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { SftpServerL3Construct, SftpServerL3ConstructProps } from '@aws-mdaa/sftp-server-l3-construct';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { SftpServerConfigParser } from './sftp-server-config';


export class SftpServerCDKApp extends MdaaCdkApp {
    constructor( props: AppProps = {} ) {
        super( props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`) )
    }
    protected subGenerateResources ( stack: Stack, l3ConstructProps: MdaaL3ConstructProps, parserProps: MdaaAppConfigParserProps ) {

        const appConfig = new SftpServerConfigParser( stack, parserProps )
        const constructProps: SftpServerL3ConstructProps = {
            ...{
                server: appConfig.server
            }, ...l3ConstructProps
        }
        new SftpServerL3Construct( stack, "server", constructProps );
        return [ stack ]
    }
}
