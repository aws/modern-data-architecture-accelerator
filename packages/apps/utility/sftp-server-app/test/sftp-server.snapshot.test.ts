/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { describe } from '@jest/globals';
import { snapShotTest, snapShotTestApp, Create } from '@aws-mdaa/testing';
import { SftpServerCDKApp } from '../lib/sftp-server';
import * as path from 'path';

describe('sftp-server Snapshot Tests', () => {
  beforeAll(() => {
    expect.addSnapshotSerializer({
      test: (val: unknown) => typeof val === 'string' && val.includes('[CONFIG:') && val.includes('test-config.yaml]'),
      print: (val: unknown) => {
        const stringVal = val as string;
        return `"${stringVal.replace(/\[CONFIG:[^[\]]*test-config\.yaml\]/, '[CONFIG:test-config.yaml]')}"`;
      },
    });
  });
  snapShotTest(
    'SFTP Stack ',
    Create.stackProvider(
      'SFTPStackMain',
      (_, context) => {
        const moduleApp = new SftpServerCDKApp({
          context: {
            ...context,
            module_configs: path.join(__dirname, 'test-config.yaml'),
          },
        });
        return moduleApp.generateStack();
      },
      {
        module_name: 'test-sftp-main',
        org: 'test-org',
        env: 'test-env',
        domain: 'test-domain',
      },
    ),
  );

  snapShotTestApp(
    'SFTP App',
    Create.appProvider(
      context => {
        const moduleApp = new SftpServerCDKApp({
          context: {
            ...context,
            module_configs: path.join(__dirname, 'test-config.yaml'),
          },
        });
        moduleApp.generateStack();
        return moduleApp;
      },
      {
        module_name: 'test-sftp-main',
        org: 'test-org',
        env: 'test-env',
        domain: 'test-domain',
      },
    ),
  );
});
