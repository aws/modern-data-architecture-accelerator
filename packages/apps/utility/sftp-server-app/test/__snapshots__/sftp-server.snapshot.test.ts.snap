// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sftp-server Snapshot Tests SFTP App App Snapshot Test 1`] = `
{
  "test-org-test-env-test-domain-test-sftp-main": {
    "Description": "(SO0320-sftp-server) Modern Data Architecture Accelerator. Version 1.0.0",
    "Outputs": {
      "REPLACED-CFN-LOGICAL-ID": {
        "Export": {
          "Name": "test-org:test-domain:test-sftp-main:server:arn",
        },
        "Value": {
          "Fn::GetAtt": [
            "REPLACED-CFN-LOGICAL-ID",
            "Arn",
          ],
        },
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Export": {
          "Name": "test-org:test-domain:test-sftp-main:server:id",
        },
        "Value": {
          "Fn::GetAtt": [
            "REPLACED-CFN-LOGICAL-ID",
            "ServerId",
          ],
        },
      },
    },
    "Parameters": {
      "BootstrapVersion": {
        "Default": "/cdk-bootstrap/hnb659fds/version",
        "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaincaefbootstraprolehelperservicetokenC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/caef-bootstrap/role-helper-service-token",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
    },
    "Resources": {
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "Stack description parameter to update on version changes",
          "Name": "/test-org/test-domain/test-sftp-main/aws-solution",
          "Tags": {
            "mdaa_cdk_app": "sftp-server",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-sftp-main",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": "(SO0320-sftp-server) Modern Data Architecture Accelerator. Version 1.0.0",
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "GroupDescription": "SFTP Transfer Service port 22",
          "GroupName": "test-org-test-env-test-domain-test-sftp-main-security-group",
          "SecurityGroupIngress": [
            {
              "CidrIp": "10.0.0.0/8",
              "FromPort": 22,
              "IpProtocol": "tcp",
              "ToPort": 22,
            },
            {
              "CidrIp": "192.168.0.0/16",
              "FromPort": 22,
              "IpProtocol": "tcp",
              "ToPort": 22,
            },
            {
              "CidrIp": "172.16.0.0/12",
              "FromPort": 22,
              "IpProtocol": "tcp",
              "ToPort": 22,
            },
          ],
          "Tags": [
            {
              "Key": "mdaa_cdk_app",
              "Value": "sftp-server",
            },
            {
              "Key": "mdaa_domain",
              "Value": "test-domain",
            },
            {
              "Key": "mdaa_env",
              "Value": "test-env",
            },
            {
              "Key": "mdaa_module_name",
              "Value": "test-sftp-main",
            },
            {
              "Key": "mdaa_org",
              "Value": "test-org",
            },
          ],
          "VpcId": "some-vpc-id",
        },
        "Type": "AWS::EC2::SecurityGroup",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "EndpointDetails": {
            "SecurityGroupIds": [
              {
                "Fn::GetAtt": [
                  "REPLACED-CFN-LOGICAL-ID",
                  "GroupId",
                ],
              },
            ],
            "SubnetIds": [
              "some-subnet-id1",
              "some-subnet-id2",
            ],
            "VpcId": "some-vpc-id",
          },
          "EndpointType": "VPC",
          "LoggingRole": {
            "Fn::GetAtt": [
              "REPLACED-CFN-LOGICAL-ID",
              "Arn",
            ],
          },
          "Protocols": [
            "SFTP",
          ],
          "SecurityPolicyName": "TransferSecurityPolicy-FIPS-2020-06",
          "Tags": [
            {
              "Key": "mdaa_cdk_app",
              "Value": "sftp-server",
            },
            {
              "Key": "mdaa_domain",
              "Value": "test-domain",
            },
            {
              "Key": "mdaa_env",
              "Value": "test-env",
            },
            {
              "Key": "mdaa_module_name",
              "Value": "test-sftp-main",
            },
            {
              "Key": "mdaa_org",
              "Value": "test-org",
            },
          ],
        },
        "Type": "AWS::Transfer::Server",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Name": "/test-org/test-domain/test-sftp-main/server/arn",
          "Tags": {
            "mdaa_cdk_app": "sftp-server",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-sftp-main",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": {
            "Fn::GetAtt": [
              "REPLACED-CFN-LOGICAL-ID",
              "Arn",
            ],
          },
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Name": "/test-org/test-domain/test-sftp-main/server/id",
          "Tags": {
            "mdaa_cdk_app": "sftp-server",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-sftp-main",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": {
            "Fn::GetAtt": [
              "REPLACED-CFN-LOGICAL-ID",
              "ServerId",
            ],
          },
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Metadata": {
          "cdk_nag": {
            "rules_to_suppress": [
              {
                "id": "AwsSolutions-IAM5",
                "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Wildcard is for log stream names, which are not known at deployment time.",
              },
              {
                "id": "NIST.800.53.R5-IAMNoInlinePolicy",
                "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
              },
              {
                "id": "HIPAA.Security-IAMNoInlinePolicy",
                "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
              },
              {
                "id": "PCI.DSS.321-IAMNoInlinePolicy",
                "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
              },
            ],
          },
        },
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": "transfer.amazonaws.com",
                },
              },
            ],
            "Version": "2012-10-17",
          },
          "RoleName": "test-org-test-env-test-domain-test-sftp-main-logging-role",
          "Tags": [
            {
              "Key": "mdaa_cdk_app",
              "Value": "sftp-server",
            },
            {
              "Key": "mdaa_domain",
              "Value": "test-domain",
            },
            {
              "Key": "mdaa_env",
              "Value": "test-env",
            },
            {
              "Key": "mdaa_module_name",
              "Value": "test-sftp-main",
            },
            {
              "Key": "mdaa_org",
              "Value": "test-org",
            },
          ],
        },
        "Type": "AWS::IAM::Role",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Metadata": {
          "cdk_nag": {
            "rules_to_suppress": [
              {
                "id": "AwsSolutions-IAM5",
                "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Wildcard is for log stream names, which are not known at deployment time.",
              },
              {
                "id": "NIST.800.53.R5-IAMNoInlinePolicy",
                "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
              },
              {
                "id": "HIPAA.Security-IAMNoInlinePolicy",
                "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
              },
              {
                "id": "PCI.DSS.321-IAMNoInlinePolicy",
                "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
              },
            ],
          },
        },
        "Properties": {
          "PolicyDocument": {
            "Statement": [
              {
                "Action": [
                  "logs:CreateLogStream",
                  "logs:DescribeLogStreams",
                  "logs:CreateLogGroup",
                  "logs:PutLogEvents",
                ],
                "Effect": "Allow",
                "Resource": [
                  {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          "Ref": "AWS::Partition",
                        },
                        ":logs:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":log-group:/aws/transfer/",
                        {
                          "Fn::GetAtt": [
                            "REPLACED-CFN-LOGICAL-ID",
                            "ServerId",
                          ],
                        },
                      ],
                    ],
                  },
                  {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          "Ref": "AWS::Partition",
                        },
                        ":logs:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":log-group:/aws/transfer/",
                        {
                          "Fn::GetAtt": [
                            "REPLACED-CFN-LOGICAL-ID",
                            "ServerId",
                          ],
                        },
                        "/*",
                      ],
                    ],
                  },
                ],
              },
            ],
            "Version": "2012-10-17",
          },
          "PolicyName": "REPLACED-CFN-LOGICAL-ID",
          "Roles": [
            {
              "Ref": "REPLACED-CFN-LOGICAL-ID",
            },
          ],
        },
        "Type": "AWS::IAM::Policy",
      },
    },
    "Rules": {
      "CheckBootstrapVersion": {
        "Assertions": [
          {
            "Assert": {
              "Fn::Not": [
                {
                  "Fn::Contains": [
                    [
                      "1",
                      "2",
                      "3",
                      "4",
                      "5",
                    ],
                    {
                      "Ref": "BootstrapVersion",
                    },
                  ],
                },
              ],
            },
            "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
          },
        ],
      },
    },
  },
}
`;

exports[`sftp-server Snapshot Tests SFTP Stack  Snapshot Test 1`] = `
{
  "Description": "(SO0320-sftp-server) Modern Data Architecture Accelerator. Version 1.0.0",
  "Outputs": {
    "REPLACED-CFN-LOGICAL-ID": {
      "Export": {
        "Name": "test-org:test-domain:test-sftp-main:server:arn",
      },
      "Value": {
        "Fn::GetAtt": [
          "REPLACED-CFN-LOGICAL-ID",
          "Arn",
        ],
      },
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Export": {
        "Name": "test-org:test-domain:test-sftp-main:server:id",
      },
      "Value": {
        "Fn::GetAtt": [
          "REPLACED-CFN-LOGICAL-ID",
          "ServerId",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaincaefbootstraprolehelperservicetokenC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/caef-bootstrap/role-helper-service-token",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "Stack description parameter to update on version changes",
        "Name": "/test-org/test-domain/test-sftp-main/aws-solution",
        "Tags": {
          "mdaa_cdk_app": "sftp-server",
          "mdaa_domain": "test-domain",
          "mdaa_env": "test-env",
          "mdaa_module_name": "test-sftp-main",
          "mdaa_org": "test-org",
        },
        "Type": "String",
        "Value": "(SO0320-sftp-server) Modern Data Architecture Accelerator. Version 1.0.0",
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "GroupDescription": "SFTP Transfer Service port 22",
        "GroupName": "test-org-test-env-test-domain-test-sftp-main-security-group",
        "SecurityGroupIngress": [
          {
            "CidrIp": "10.0.0.0/8",
            "FromPort": 22,
            "IpProtocol": "tcp",
            "ToPort": 22,
          },
          {
            "CidrIp": "192.168.0.0/16",
            "FromPort": 22,
            "IpProtocol": "tcp",
            "ToPort": 22,
          },
          {
            "CidrIp": "172.16.0.0/12",
            "FromPort": 22,
            "IpProtocol": "tcp",
            "ToPort": 22,
          },
        ],
        "Tags": [
          {
            "Key": "mdaa_cdk_app",
            "Value": "sftp-server",
          },
          {
            "Key": "mdaa_domain",
            "Value": "test-domain",
          },
          {
            "Key": "mdaa_env",
            "Value": "test-env",
          },
          {
            "Key": "mdaa_module_name",
            "Value": "test-sftp-main",
          },
          {
            "Key": "mdaa_org",
            "Value": "test-org",
          },
        ],
        "VpcId": "some-vpc-id",
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "EndpointDetails": {
          "SecurityGroupIds": [
            {
              "Fn::GetAtt": [
                "REPLACED-CFN-LOGICAL-ID",
                "GroupId",
              ],
            },
          ],
          "SubnetIds": [
            "some-subnet-id1",
            "some-subnet-id2",
          ],
          "VpcId": "some-vpc-id",
        },
        "EndpointType": "VPC",
        "LoggingRole": {
          "Fn::GetAtt": [
            "REPLACED-CFN-LOGICAL-ID",
            "Arn",
          ],
        },
        "Protocols": [
          "SFTP",
        ],
        "SecurityPolicyName": "TransferSecurityPolicy-FIPS-2020-06",
        "Tags": [
          {
            "Key": "mdaa_cdk_app",
            "Value": "sftp-server",
          },
          {
            "Key": "mdaa_domain",
            "Value": "test-domain",
          },
          {
            "Key": "mdaa_env",
            "Value": "test-env",
          },
          {
            "Key": "mdaa_module_name",
            "Value": "test-sftp-main",
          },
          {
            "Key": "mdaa_org",
            "Value": "test-org",
          },
        ],
      },
      "Type": "AWS::Transfer::Server",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Name": "/test-org/test-domain/test-sftp-main/server/arn",
        "Tags": {
          "mdaa_cdk_app": "sftp-server",
          "mdaa_domain": "test-domain",
          "mdaa_env": "test-env",
          "mdaa_module_name": "test-sftp-main",
          "mdaa_org": "test-org",
        },
        "Type": "String",
        "Value": {
          "Fn::GetAtt": [
            "REPLACED-CFN-LOGICAL-ID",
            "Arn",
          ],
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Name": "/test-org/test-domain/test-sftp-main/server/id",
        "Tags": {
          "mdaa_cdk_app": "sftp-server",
          "mdaa_domain": "test-domain",
          "mdaa_env": "test-env",
          "mdaa_module_name": "test-sftp-main",
          "mdaa_org": "test-org",
        },
        "Type": "String",
        "Value": {
          "Fn::GetAtt": [
            "REPLACED-CFN-LOGICAL-ID",
            "ServerId",
          ],
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-IAM5",
              "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Wildcard is for log stream names, which are not known at deployment time.",
            },
            {
              "id": "NIST.800.53.R5-IAMNoInlinePolicy",
              "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
            },
            {
              "id": "HIPAA.Security-IAMNoInlinePolicy",
              "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
            },
            {
              "id": "PCI.DSS.321-IAMNoInlinePolicy",
              "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
            },
          ],
        },
      },
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "transfer.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "RoleName": "test-org-test-env-test-domain-test-sftp-main-logging-role",
        "Tags": [
          {
            "Key": "mdaa_cdk_app",
            "Value": "sftp-server",
          },
          {
            "Key": "mdaa_domain",
            "Value": "test-domain",
          },
          {
            "Key": "mdaa_env",
            "Value": "test-env",
          },
          {
            "Key": "mdaa_module_name",
            "Value": "test-sftp-main",
          },
          {
            "Key": "mdaa_org",
            "Value": "test-org",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Metadata": {
        "cdk_nag": {
          "rules_to_suppress": [
            {
              "id": "AwsSolutions-IAM5",
              "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Wildcard is for log stream names, which are not known at deployment time.",
            },
            {
              "id": "NIST.800.53.R5-IAMNoInlinePolicy",
              "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
            },
            {
              "id": "HIPAA.Security-IAMNoInlinePolicy",
              "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
            },
            {
              "id": "PCI.DSS.321-IAMNoInlinePolicy",
              "reason": "[MDAA:@aws-mdaa/utility/sftp-server-l3-construct/lib/sftp-server-l3-construct.ts:106:25] Role is specific to this server. Inline policy is appropriate.",
            },
          ],
        },
      },
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogStream",
                "logs:DescribeLogStreams",
                "logs:CreateLogGroup",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/transfer/",
                      {
                        "Fn::GetAtt": [
                          "REPLACED-CFN-LOGICAL-ID",
                          "ServerId",
                        ],
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/transfer/",
                      {
                        "Fn::GetAtt": [
                          "REPLACED-CFN-LOGICAL-ID",
                          "ServerId",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "REPLACED-CFN-LOGICAL-ID",
        "Roles": [
          {
            "Ref": "REPLACED-CFN-LOGICAL-ID",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
